geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/membership/GMSJoinLeave.java

```diff
@@ -1454,10 +1454,14 @@ void processMessage(NetworkPartitionMessage<ID> msg) {
     if (isStopping) {
       return;
     }
-
-    String str = "Membership coordinator " + msg.getSender()
-        + " has declared that a network partition has occurred";
-    forceDisconnect(str);
+    ID sender = msg.getSender();
+    if (getView().getMembers().contains(sender) || getView().getViewId() == sender.getVmViewId()) {
```

Comment:
I don't understand the reason for the viewId check you've added here.  Would you please elaborate?

Reply:
Thanks for pointing this out, @bschuchardt. I removed the check.